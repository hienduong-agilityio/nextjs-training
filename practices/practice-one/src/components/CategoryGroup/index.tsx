'use client';

// Libraries
import { usePathname, useSearchParams } from 'next/navigation';
import { twMerge } from 'tailwind-merge';

// Components
import Link from 'next/link';

// Helpers
import { capitalizeText } from '@/helpers';

// Constants
import { ALL_CATEGORIES, SEARCH_PARAMS } from '@/constants';

interface ICategoryGroupProps {
  title: string;
  items: { name: string; count: number }[];
}

export function CategoryGroup({ title, items }: ICategoryGroupProps) {
  const pathname = usePathname();
  const searchParams = useSearchParams();

  const currentCategory =
    searchParams.get(SEARCH_PARAMS.CATEGORY) ?? ALL_CATEGORIES.ALL;

  const handleCategoryClick = (itemName: string) => {
    const updatedParams = new URLSearchParams();

    // Update the category query parameter
    if (itemName !== ALL_CATEGORIES.ALL) {
      updatedParams.set(SEARCH_PARAMS.CATEGORY, itemName);
    }

    return `${pathname}?${updatedParams.toString()}`;
  };

  return (
    <div className="mb-6 bg-secondary-100 p-4 rounded">
      <h3 className="text-lg font-bold mb-4">{title}</h3>
      <ul className="space-y-2 text-sm text-gray-700">
        {items.map((item) => {
          const isActive =
            item.name === ALL_CATEGORIES.ALL
              ? currentCategory === ALL_CATEGORIES.ALL || !currentCategory
              : item.name === currentCategory;

          return (
            <li className="flex justify-between" key={item.name}>
              <Link
                href={handleCategoryClick(item.name)}
                className={twMerge(
                  'w-full block font-semibold py-2 rounded hover:bg-secondary-200',
                  isActive
                    ? 'bg-primary-500 text-primary-200'
                    : 'text-gray-800',
                )}
              >
                {capitalizeText(item.name)}
              </Link>
            </li>
          );
        })}
      </ul>
    </div>
  );
}
