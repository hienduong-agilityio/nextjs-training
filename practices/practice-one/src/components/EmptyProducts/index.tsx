import Link from 'next/link';

export const EmptyProducts = ({
  message,
  actionLabel,
  actionHref,
}: {
  message: string;
  actionLabel: string;
  actionHref: string;
}) => (
  <div className="flex flex-col items-center gap-4 py-16">
    <p className="text-xl text-gray-600">{message}</p>
    <Link
      href={actionHref}
      className="inline-block px-6 py-2 text-white bg-blue-500 rounded-lg"
    >
      {actionLabel}
    </Link>
  </div>
);
