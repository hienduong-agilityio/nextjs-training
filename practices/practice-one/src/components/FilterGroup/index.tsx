interface FilterGroupProps {
  title: string;
  items: { name: string; count: number }[];
}

export const FilterGroup = ({ title, items }: FilterGroupProps) => {
  return (
    <div className="mb-6 bg-secondary-100 p-4 rounded">
      <h3 className="text-lg font-bold mb-4">{title}</h3>
      <ul className="space-y-2 text-sm text-gray-700">
        {items.map((item) => (
          <li key={item.name} className="flex justify-between py-2">
            <button className="font-semibold hover:text-primary-200 ">
              {item.name}
            </button>
            <span>({item.count})</span>
          </li>
        ))}
      </ul>
    </div>
  );
};
