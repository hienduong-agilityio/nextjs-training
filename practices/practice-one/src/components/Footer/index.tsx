// Icon
import {
  FacebookIcon,
  LogoIcon,
  MasterCardIcon,
  PaypalIcon,
  TwitterIcon,
  VisaIcon,
  WesternIcon,
} from '@/icons';

// Mock
import { FOOTER_NAVIGATION } from '@/mocks';

// Components
import Link from 'next/link';

// Constants
import { ROUTE } from '@/constants';

export const Footer = () => {
  return (
    <footer className="w-full bg-primary-50">
      <div className="md:container px-4 pt-6 md:mx-auto sm:px-6 md:px-8 lg:px-12 xl:px-16 sm:pt-8 md:pt-12 lg:pt-16 xl:pt-20">
        <div className="py-8">
          {/* Main Footer Content */}
          <div className="flex flex-col gap-10 lg:flex-row lg:justify-between">
            {/* Logo Section */}
            <div className="max-w-[290px] flex flex-col gap-4">
              <Link href={ROUTE.PRODUCTS} className="flex items-center gap-2">
                <LogoIcon aria-label="logo" />
                <span className="text-lg font-bold sm:text-xl lg:text-2xl">
                  E-Comm
                </span>
              </Link>
              <span className="text-sm font-normal leading-normal">
                Lorem Ipsum is simply dummy text of the printing and typesetting
                industry. Lorem Ipsum has been the industry standard dummy text
                ever. Since the 1500s, when an unknown printer.
              </span>
            </div>

            <div className="flex flex-col w-full gap-8 sm:flex-row sm:justify-evenly lg:w-3/4">
              {/* Follow Us */}
              <div className="lg:max-w-[220px] flex flex-col gap-4">
                <h4 className="font-medium text-foreground-400">Follow Us</h4>
                <span className="text-sm">
                  Since the 1500s, when an unknown printer took a galley of type
                  and scrambled.
                </span>
                <div className="flex gap-4">
                  <Link href="https://www.facebook.com/" aria-label="Facebook">
                    <FacebookIcon color="#385C8E" />
                  </Link>
                  <Link href="https://www.twitter.com/" aria-label="Twitter">
                    <TwitterIcon />
                  </Link>
                </div>
              </div>

              {/* Contact Us */}
              <div className="lg:max-w-[150px] flex flex-col gap-4">
                <h4 className="font-medium text-foreground-400">Contact Us</h4>
                <address className="text-sm not-italic">
                  E-Comm, 4578 Marmora Road, Glasgow D04 89GR
                </address>
              </div>
            </div>
          </div>

          {/* Footer Links */}
          <div className="grid grid-cols-1 gap-8 pt-16 sm:grid-cols-2 lg:grid-cols-4">
            {FOOTER_NAVIGATION.map((item) => (
              <ul key={item.title}>
                <li>
                  <h4 className="mb-5 font-medium text-foreground-400">
                    {item.title}
                  </h4>
                  <ul>
                    {item.items.map((item) => (
                      <li
                        key={item.name}
                        className="mb-3 text-sm text-content2 hover:text-primary-700"
                      >
                        <Link href={item.url} className="hover:underline">
                          {item.name}
                        </Link>
                      </li>
                    ))}
                  </ul>
                </li>
              </ul>
            ))}
          </div>
        </div>

        {/* Footer Bottom */}
        <div className="flex flex-col items-center justify-between gap-6 py-6 border-t-2 border-secondary-100 lg:flex-row ">
          <p className="text-sm text-secondary-500">
            &#169; 2018 Ecommerce theme by www.bisenbaev.com
          </p>
          <div className="flex gap-2">
            <WesternIcon aria-label="Western" />
            <MasterCardIcon aria-label="MasterCard" />
            <PaypalIcon aria-label="Paypal" />
            <VisaIcon aria-label="Visa" />
          </div>
        </div>
      </div>
    </footer>
  );
};
