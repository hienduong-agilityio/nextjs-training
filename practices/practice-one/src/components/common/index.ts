export { default as Button } from './Button';
export { default as InputField } from './InputField';
export { default as Tabs } from './Tabs';
export { default as Select } from './Select';
export { default as IconButton } from './IconButton';
export * from './IconButton';
