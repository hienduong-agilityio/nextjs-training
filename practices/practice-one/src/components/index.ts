export * from './Header';
export { default as SearchBox } from './SearchBox';
export * from './SearchBox';
export * from './ProductTab';

// Common
export * from './common';
