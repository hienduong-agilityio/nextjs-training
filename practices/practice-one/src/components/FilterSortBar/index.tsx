'use client';

// Hooks
import { useSearchParams, useRouter } from 'next/navigation';
import { useEffect, useState } from 'react';

// Components
import { Select } from '@/components';

// Constants
import { SEARCH_PARAMS, SORT_PRODUCT_OPTIONS } from '@/constants';

interface FilterSortBarProps {
  itemCount: number;
  sortOptions: string[];
  showOptions: string[];
}

export const FilterSortBar = ({
  itemCount = 0,
  sortOptions = SORT_PRODUCT_OPTIONS,
  showOptions = ['6'],
}: FilterSortBarProps) => {
  const searchParams = useSearchParams();
  const router = useRouter();

  const [selectedLimit, setSelectedLimit] = useState<string>(showOptions[0]);
  const [selectedSort, setSelectedSort] = useState<string>(sortOptions[0]);

  // TODO: Remove useEffect.
  useEffect(() => {
    // Get query parameters from the URL
    const limitParam = searchParams.get(SEARCH_PARAMS.LIMIT) ?? showOptions[0];
    const sortParam = searchParams.get(SEARCH_PARAMS.SORT_BY) ?? sortOptions[0];

    setSelectedLimit(limitParam);
    setSelectedSort(sortParam);
  }, [searchParams, showOptions, sortOptions]);

  const updateQueryParams = (param: string, value: string) => {
    const currentParams = new URLSearchParams(searchParams.toString());
    currentParams.set(param, value);

    // Reset pagination if a filter changes
    if (param === SEARCH_PARAMS.LIMIT) {
      currentParams.delete(SEARCH_PARAMS.PAGE);
    }

    router.push(`?${currentParams.toString()}`);
  };

  const handleShowChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    const value = event.target.value;

    setSelectedLimit(value);
    updateQueryParams(SEARCH_PARAMS.LIMIT, value);
  };

  const handleSortChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    const value = event.target.value;

    setSelectedSort(value);
    updateQueryParams(SEARCH_PARAMS.SORT_BY, value);
  };

  return (
    <div className="bg-secondary-100 p-4 shadow-md md:sticky rounded top-0 z-30 w-full">
      <div className="flex flex-col sm:flex-row sm:justify-between sm:items-center space-y-4 sm:space-y-0">
        {/* Item count */}
        <div className="text-sm w-max text-gray-600">{itemCount} Items</div>

        {/* Filters */}
        <div className="flex flex-col sm:flex-row sm:items-center sm:space-x-4 space-y-4 sm:space-y-0">
          <div className="flex gap-2 items-center">
            <span>Sort</span>
            {/* TODO: Custom style for select arrow by tailwinds classes*/}
            <Select
              options={sortOptions}
              value={selectedSort}
              customClass="w-auto bg-secondary-100 capitalize"
              onChange={handleSortChange}
            />
          </div>
          <div className="flex gap-2 items-center">
            <span>Show</span>
            <Select
              options={showOptions}
              value={selectedLimit}
              customClass="w-auto bg-secondary-100"
              onChange={handleShowChange}
            />
          </div>
        </div>
      </div>
    </div>
  );
};
