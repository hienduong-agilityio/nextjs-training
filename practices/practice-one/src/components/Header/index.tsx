// Icon
import { CartIcon, LogoIcon, ProfileIcon } from '@/icons';

// Components
import { SearchBox } from '@/components';

interface IHeaderProps {
  cartItemCount?: number;
}

export const Header = ({ cartItemCount = 0 }: IHeaderProps) => {
  return (
    <header className="container flex flex-wrap items-center w-full px-4 py-4 pb-6 mx-auto md:gap-8 md:justify-start xl:justify-between lg:pb-12 sm:px-6 md:px-8 lg:px-12 xl:px-16 sm:py-6 md:py-8 lg:py-10 xl:py-12">
      <div className="flex flex-wrap gap-5 xl:justify-between lg:w-auto">
        <a href="/" className="flex items-center gap-2">
          {/* Logo */}
          <LogoIcon aria-label="logo" />
          <span className="text-lg font-bold sm:text-xl lg:text-2xl">
            E-Comm
          </span>
        </a>

        {/* Search Box */}
        <SearchBox
          customClass={{
            container: 'md:justify-end',
            input: 'w-full md:w-[400px] lg:w-[520px]',
            inputContainer: 'border-primary-300',
            button: 'bg-primary-300 text-white',
          }}
        />
      </div>

      <div className="flex items-center gap-4 mt-5 sm:gap-6 lg:mt-0">
        {/* Profile */}
        <div className="flex items-center gap-2 text-sm sm:text-base lg:text-lg">
          <ProfileIcon
            aria-label="profile"
            className="flex items-center"
            size={20}
          />
          <span>My account</span>
        </div>

        {/* Cart */}
        <div className="relative flex items-center">
          <CartIcon aria-label="cart" size={25} itemCount={cartItemCount} />
        </div>
      </div>
    </header>
  );
};
