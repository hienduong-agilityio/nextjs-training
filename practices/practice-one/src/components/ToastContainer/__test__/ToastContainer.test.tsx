import { act, render, screen, waitFor, cleanup } from '@testing-library/react';

// Stores
import { ToastStore } from '@/stores';

// Components
import { ToastContainer } from '@/components';

// Constants
import { STATUS_TYPES, TOAST_MESSAGES } from '@/constants';

describe('ToastContainer and ToastStore', () => {
  const DEFAULT_MESSAGE = TOAST_MESSAGES.API_ERROR;

  const setupToast = (
    message: string,
    type: string,
    timeoutDuration: number = 3000,
  ) => {
    act(() => {
      ToastStore.getState().showToast(message, type, timeoutDuration);
    });
    return render(<ToastContainer />);
  };

  afterEach(() => {
    cleanup();
    act(() => {
      ToastStore.setState({
        message: '',
        type: STATUS_TYPES.SUCCESS,
        isVisible: false,
        timeoutDuration: 3000,
      });
    });
  });

  it('should render the Toast when `isVisible` is true', async () => {
    const message = 'This is a success message';
    const { asFragment } = setupToast(message, STATUS_TYPES.SUCCESS);

    const toastElement = screen.getByLabelText('toast success');
    expect(toastElement).toBeInTheDocument();
    expect(screen.getByText(message)).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('should hide the Toast after the specified timeout duration', async () => {
    const message = 'This toast will disappear';
    const timeoutDuration = 1000;

    const { asFragment } = setupToast(
      message,
      STATUS_TYPES.SUCCESS,
      timeoutDuration,
    );

    expect(screen.getByText(message)).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();

    await waitFor(
      () => {
        expect(screen.queryByText(message)).not.toBeInTheDocument();
      },
      { timeout: timeoutDuration + 500 },
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display different Toast types correctly', () => {
    const messages = [
      {
        message: 'Success',
        type: STATUS_TYPES.SUCCESS,
        label: 'toast success',
      },
      {
        message: 'Warning',
        type: STATUS_TYPES.WARNING,
        label: 'toast warning',
      },
      { message: 'Error', type: STATUS_TYPES.ERROR, label: 'toast error' },
    ];

    messages.forEach(({ message, type, label }) => {
      const { asFragment } = setupToast(message, type);
      const toastElements = screen.getAllByLabelText(label);

      expect(toastElements.length).toBe(1);
      expect(screen.getByText(message)).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
      cleanup();
    });
  });

  it('should reset the store when `hideToast` is called', () => {
    const { hideToast, showToast } = ToastStore.getState();

    act(() => {
      showToast(DEFAULT_MESSAGE, STATUS_TYPES.SUCCESS);
    });
    expect(ToastStore.getState().isVisible).toBe(true);

    act(() => {
      hideToast();
    });
    expect(ToastStore.getState().isVisible).toBe(false);
    expect(ToastStore.getState().message).toBe(DEFAULT_MESSAGE);
  });
});
