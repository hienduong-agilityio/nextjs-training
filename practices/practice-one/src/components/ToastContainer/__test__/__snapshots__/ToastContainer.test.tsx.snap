// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastContainer and ToastStore should display different Toast types correctly 1`] = `
<DocumentFragment>
  <div
    class="fixed top-9 z-40 left-1/2 transform -translate-x-1/2 w-full max-w-[calc(100%-2rem)] flex justify-center"
  >
    <div
      aria-label="toast success"
      class="flex items-center divide-x rtl:divide-x-reverse px-4 py-3 rounded-lg shadow transition-transform duration-300 w-max max-w-full text-white bg-success divide-green-950"
    >
      <div
        class="text-sm font-normal inline-flex items-center gap-3"
      >
        <svg
          fill="none"
          height="12"
          viewBox="0 0 12 12"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M6.0022 12C9.31591 12 12.0022 9.31371 12.0022 6C12.0022 2.68629 9.31591 0 6.0022 0C2.68849 0 0.00219727 2.68629 0.00219727 6C0.00219727 9.31371 2.68849 12 6.0022 12ZM7.94379 3.91404C8.17067 3.67328 8.59119 3.71853 8.82444 3.91404C9.08494 4.13238 9.03673 4.49289 8.82444 4.71814L5.65234 8.08426C5.44861 8.30045 4.97439 8.30998 4.77167 8.08426L3.17782 6.30949C2.95656 6.06311 2.9311 5.75078 3.17782 5.50538C3.39002 5.29432 3.85106 5.27441 4.05848 5.50538L5.22261 6.80165L7.94379 3.91404Z"
            fill="#A9EBCA"
            fill-rule="evenodd"
          />
        </svg>
        Success
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ToastContainer and ToastStore should display different Toast types correctly 2`] = `
<DocumentFragment>
  <div
    class="fixed top-9 z-40 left-1/2 transform -translate-x-1/2 w-full max-w-[calc(100%-2rem)] flex justify-center"
  >
    <div
      aria-label="toast warning"
      class="flex items-center divide-x rtl:divide-x-reverse px-4 py-3 rounded-lg shadow transition-transform duration-300 w-max max-w-full"
    >
      <div
        class="text-sm font-normal inline-flex items-center gap-3"
      >
        <svg
          height="32"
          style="overflow: visible;"
          viewBox="0 0 32 32"
          width="32"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g>
            <circle
              cx="16"
              cy="16"
              fill="#D72828"
              r="16"
            />
            <path
              d="M14.5,25h3v-3h-3V25z M14.5,6v13h3V6H14.5z"
              fill="#E6E6E6"
            />
          </g>
        </svg>
        Warning
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ToastContainer and ToastStore should display different Toast types correctly 3`] = `
<DocumentFragment>
  <div
    class="fixed top-9 z-40 left-1/2 transform -translate-x-1/2 w-full max-w-[calc(100%-2rem)] flex justify-center"
  >
    <div
      aria-label="toast error"
      class="flex items-center divide-x rtl:divide-x-reverse px-4 py-3 rounded-lg shadow transition-transform duration-300 w-max max-w-full text-red-100 bg-red-800 divide-red-950"
    >
      <div
        class="text-sm font-normal inline-flex items-center gap-3"
      >
        <svg
          height="32"
          style="overflow: visible;"
          viewBox="0 0 32 32"
          width="32"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g>
            <circle
              cx="16"
              cy="16"
              fill="#D72828"
              r="16"
            />
            <path
              d="M14.5,25h3v-3h-3V25z M14.5,6v13h3V6H14.5z"
              fill="#E6E6E6"
            />
          </g>
        </svg>
        Error
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ToastContainer and ToastStore should hide the Toast after the specified timeout duration 1`] = `
<DocumentFragment>
  <div
    class="fixed top-9 z-40 left-1/2 transform -translate-x-1/2 w-full max-w-[calc(100%-2rem)] flex justify-center"
  >
    <div
      aria-label="toast success"
      class="flex items-center divide-x rtl:divide-x-reverse px-4 py-3 rounded-lg shadow transition-transform duration-300 w-max max-w-full text-white bg-success divide-green-950"
    >
      <div
        class="text-sm font-normal inline-flex items-center gap-3"
      >
        <svg
          fill="none"
          height="12"
          viewBox="0 0 12 12"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M6.0022 12C9.31591 12 12.0022 9.31371 12.0022 6C12.0022 2.68629 9.31591 0 6.0022 0C2.68849 0 0.00219727 2.68629 0.00219727 6C0.00219727 9.31371 2.68849 12 6.0022 12ZM7.94379 3.91404C8.17067 3.67328 8.59119 3.71853 8.82444 3.91404C9.08494 4.13238 9.03673 4.49289 8.82444 4.71814L5.65234 8.08426C5.44861 8.30045 4.97439 8.30998 4.77167 8.08426L3.17782 6.30949C2.95656 6.06311 2.9311 5.75078 3.17782 5.50538C3.39002 5.29432 3.85106 5.27441 4.05848 5.50538L5.22261 6.80165L7.94379 3.91404Z"
            fill="#A9EBCA"
            fill-rule="evenodd"
          />
        </svg>
        This toast will disappear
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ToastContainer and ToastStore should hide the Toast after the specified timeout duration 2`] = `<DocumentFragment />`;

exports[`ToastContainer and ToastStore should render the Toast when \`isVisible\` is true 1`] = `
<DocumentFragment>
  <div
    class="fixed top-9 z-40 left-1/2 transform -translate-x-1/2 w-full max-w-[calc(100%-2rem)] flex justify-center"
  >
    <div
      aria-label="toast success"
      class="flex items-center divide-x rtl:divide-x-reverse px-4 py-3 rounded-lg shadow transition-transform duration-300 w-max max-w-full text-white bg-success divide-green-950"
    >
      <div
        class="text-sm font-normal inline-flex items-center gap-3"
      >
        <svg
          fill="none"
          height="12"
          viewBox="0 0 12 12"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M6.0022 12C9.31591 12 12.0022 9.31371 12.0022 6C12.0022 2.68629 9.31591 0 6.0022 0C2.68849 0 0.00219727 2.68629 0.00219727 6C0.00219727 9.31371 2.68849 12 6.0022 12ZM7.94379 3.91404C8.17067 3.67328 8.59119 3.71853 8.82444 3.91404C9.08494 4.13238 9.03673 4.49289 8.82444 4.71814L5.65234 8.08426C5.44861 8.30045 4.97439 8.30998 4.77167 8.08426L3.17782 6.30949C2.95656 6.06311 2.9311 5.75078 3.17782 5.50538C3.39002 5.29432 3.85106 5.27441 4.05848 5.50538L5.22261 6.80165L7.94379 3.91404Z"
            fill="#A9EBCA"
            fill-rule="evenodd"
          />
        </svg>
        This is a success message
      </div>
    </div>
  </div>
</DocumentFragment>
`;
